### PR TITLE
move all meck handling into init/end_per_testcase

### PR DESCRIPTION
**Pull request by RoadRunnr**
Invoking meck:unload from the test case itself bears the risk of
not executing it when the test case failes.

### CODE REVIEW

File: test/ggsn_proxy_SUITE.erl [hunk 1]
```diff
@@ -835,6 +843,14 @@ end_per_testcase(delete_bearer_requests_multi, Config) ->
...
     end_per_testcase(Config),
     Config;
+end_per_testcase(TestCase, Config)
+  when TestCase == proxy_context_selection;
+       TestCase == proxy_context_invalid_selection;
+       TestCase == proxy_context_invalid_mapping;
+       TestCase == proxy_api_v2 ->
+    ok = meck:unload(gtp_proxy_ds),
+    end_per_testcase(Config),
+    Config;
```
Comment (ebengt) [inline test/ggsn_proxy_SUITE.erl:853]: I think the return value is ignored (apart from special tagged tuples). So just 'ok' is a good return value.

**Review by ebengt — APPROVED** [review]

File: test/pgw_proxy_SUITE.erl [hunk 2]
```diff
@@ -895,10 +907,22 @@ end_per_testcase(dns_node_selection, Config) ->
     ok = meck:unload(inet_res),
     end_per_testcase(Config),
     Config;
+end_per_testcase(sx_upf_removal, Config) ->
+    ok = meck:unload(ergw_sx_node),
```
Comment (ebengt) [inline test/pgw_proxy_SUITE.erl:911]: This was not done in previous version.
For symmetry reason it is good that it is added, but might there have been a reason not to do it before?

**Review by ebengt — APPROVED** [review]

**Review by vkatsuba — APPROVED** [review]

File: test/ggsn_proxy_SUITE.erl (hunk 1 above)
Comment (RoadRunnr) [reply test/ggsn_proxy_SUITE.erl:853]: yes, but all end_per_testcase function currently do return config. No need to change that now.

File: test/pgw_proxy_SUITE.erl (hunk 2 above)
Comment (RoadRunnr) [reply test/pgw_proxy_SUITE.erl:911]: the missing unload was actually causing test failures and was the reason I looked at this.

**Pull request merged**